app/queues/dns/add-record-flow.server.ts

```diff
@@ -0,0 +1,103 @@
+import { FlowProducer } from 'bullmq';
+import invariant from 'tiny-invariant';
+import { redis } from '~/lib/redis.server';
+import { buildDomain } from '~/utils';
+import { dnsUpdateQueueName } from './workers/dns-update-worker.server';
+import { checkDnsStatusQueueName } from './workers/poll-dns-status-worker.server';
+import { syncDbStatusQueueName } from './workers/sync-db-status-worker.server';
+import { createUserRecord } from '~/models/record.server';
+
+import type { FlowJob } from 'bullmq';
+import type { Record } from '@prisma/client';
+import type { DnsUpdaterData } from './workers/dns-update-worker.server';
+import type { DbRecordSynchronizerData } from './workers/sync-db-status-worker.server';
+
+export type FqdnOrSubdomain = { subdomain: string } | { fqdn: string };
+interface BaseAddDnsRequestData {
```

Comment:
Can we use pick here too? Genuine question, I really don't know because of the optional attributes